sound/soc/sof/intel/hda-loader.c

```diff
@@ -364,7 +365,8 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 		dev_dbg(sdev->dev,
 			"Attempting iteration %d of Core En/ROM load...\n", i);
 
-		ret = cl_dsp_init(sdev, stream->hstream.stream_tag, i + 1);
+		hda->boot_iteration = i;
```

Comment:
@plbossart this should be i + 1 for the conditions if (hda->boot_iteration == HDA_FW_BOOT_ATTEMPTS) to be true.

Reply:
fixed